File: presentation/src/main/java/cleanarcpro/brightowusu/com/cleanarcproj/di/modules/ActivityUserModule.java
package cleanarcpro.brightowusu.com.cleanarcproj.di.modules;

import android.arch.lifecycle.ViewModelProviders;

import cleanarcpro.brightowusu.com.cleanarcproj.data.repository.UserRepositoryImpl;
import cleanarcpro.brightowusu.com.cleanarcproj.di.ActivityScope;
import cleanarcpro.brightowusu.com.cleanarcproj.domain.interactors.GetUserUseCase;
import cleanarcpro.brightowusu.com.cleanarcproj.domain.interactors.IGetUserInteractor;
import cleanarcpro.brightowusu.com.cleanarcproj.domain.models.DomainUser;
import cleanarcpro.brightowusu.com.cleanarcproj.view.ActivityUser;
import cleanarcpro.brightowusu.com.cleanarcproj.viewmodels.DisplayUserViewModel;
import dagger.Module;
import dagger.Provides;
import rx.Observable;

//@Module
public class ActivityUserModule {

    private final ActivityUser activityUser;

    public ActivityUserModule(ActivityUser activityUser) {
        this.activityUser = activityUser;
    }

    //@Provides
    //@ActivityScope
    public DisplayUserViewModel providesDisplayUserViewModel(final IGetUserInteractor getUserInteractor) {

        DisplayUserViewModel displayUserViewModel
                = ViewModelProviders.of(activityUser).get(DisplayUserViewModel.class);

        displayUserViewModel.setUserInteractor(getUserInteractor);
        return displayUserViewModel;
    }
}

File: presentation/src/main/java/cleanarcpro/brightowusu/com/cleanarcproj/di/modules/ActivityUserModule.kt
package cleanarcpro.brightowusu.com.cleanarcproj.di.modules;

import android.arch.lifecycle.ViewModelProviders;

import cleanarcpro.brightowusu.com.cleanarcproj.di.ActivityScope;
import cleanarcpro.brightowusu.com.cleanarcproj.domain.interactors.IGetUserInteractor;
import cleanarcpro.brightowusu.com.cleanarcproj.view.activities.ActivityHome;
import cleanarcpro.brightowusu.com.cleanarcproj.viewmodels.DisplayUserViewModel;
import dagger.Module;
import dagger.Provides;

//@Module
class ActivityUserModule(val activityHome: ActivityHome) {

//    @Provides
//    @ActivityScope
//    fun providesDisplayUserViewModel(getUserInteractor: IGetUserInteractor ): DisplayUserViewModel{
//
//        val displayUserViewModel
//                = ViewModelProviders.of(activityHome).get(DisplayUserViewModel::class.java)
//
//        displayUserViewModel.userInteractor = getUserInteractor
//        return displayUserViewModel
//
//    }
}
